src/transports/http.c

```diff
@@ -359,6 +360,9 @@ static int on_headers_complete(http_parser *parser)
 					t->parse_error = PARSE_ERROR_REPLAY;
 					return 0;
 				}
+			} else {
+				giterr_set(GITERR_NET, "authentication failed, incorrect credentials");
```

Comment:
The Git server doesn't do much here, this comes typically from the HTTP in front of it. But 403 quite explicitly does not mean that authentication failed. It means it succeeded but authorisation failed.